File: Code/3DMANAGER-APP/3dmanager-app.client/src/models/filament/FilamentDetailObject.ts
import type { FileResponse } from "../file/fileResponse";

export interface FilamentDetailObject {
    groupId: number;
    filamentId: number;
    filamentName: string;
    filamentType: string;
    filamentWeight: number;
    filamentColor: string;
    filamentTemperature: number;
    filamentLenght: number;
    filamentRemainingLenght: number;
    filamentThickness: number;
    filamentCost: number;
    filamentDescription: string;
    filamentCreateDate: Date;
    filamentState: number;
    filamentPrintedPrintsMonth: number;
    filamentPrintedPrintsTotal: number;
    filamentImageFile?: FileResponse | null;
}

File: Code/3DMANAGER-APP/3DMANAGER-APP.BLL/Models/Filament/FilamentDetailObject.cs
using _3DMANAGER_APP.BLL.Models.File;

namespace _3DMANAGER_APP.BLL.Models.Filament
{
    public class FilamentDetailObject
    {
        public int FilamentId { get; set; }
        public string FilamentName { get; set; }
        public string FilamentType { get; set; }
        public int FilamentWeight { get; set; }
        public string FilamentColor { get; set; }
        public int FilamentTemperature { get; set; }
        public decimal FilamentLenght { get; set; }
        public decimal FilamentRemainingLenght { get; set; }
        public float FilamentThickness { get; set; }
        public decimal FilamentCost { get; set; }
        public string FilamentDescription { get; set; }
        public DateTime FilamentCreateDate { get; set; }
        public int FilamentState { get; set; }
        public int FilamentPrintedPrintsMonth { get; set; }
        public int FilamentPrintedPrintsTotal { get; set; }
        public FileResponse? FilamentImageFile { get; set; }

    }
}
